roadmap/implementers-guide/src/runtime/disputes2.md

```diff
@@ -0,0 +1,57 @@
+
+At this point the validator set can not be trusted (since they voted for the block despite something being
+fishy at the very least) or the validator that blew the whistle has ulterior motives to do so (i.e. it is controlled by a third party and wants to incur damage to itself).
+In either way, there are malicious validators around.
+As a consequence, all validators at the time of block backing, are being notified via broadcast of
+the first challenging vote.
+Validators that backed the candidate implicitly voted for backing the candidate and may not vote a second time in different fashion, i.e. a pro and a con vote would be punished, two positive votes would be fine, but the first would be counted.
+Two opposing votes by the same validator would be equal to an attempted double vote and would be slashed accordingly.
+
+All validators at block inclusion time are eligible to (and should) cast their Vote. The backing votes of backing checkers
+are counted as votes as well.
+
+## Initiation
+
+A dispute is initiated by one approval checker creating and gossiping a vote, that challenges the vote.
+approval checkers run the validation code and use this to verify the transactions.
+
+After a approval checker challenged a block, all validators that received the gossiped vote, reconstruct the block
+from availability erasure code chunks and check the block themselves via the validation code.
+The result of that check is converted into a vote, and distributed via the same mechanics as the first one.
+
+Once a receiver receives quorum and supermajority in one or the other direction, the
+vote is concluded.
+Conclusion implies that the result for this block can not be altered anymore, valid or invalid is fixed now.
+
+In order to assure, the dispute result is not forgotten or intentionally side stepped, it has to be recorded on chain.
+This on chain recording mechanic must be vigilant, in a sense, that new emerging forks
+must also receive the dispute resolution recorded (transplantation) iff the disputed block is in the
+ancestry path of that chain.
+
+If the disputed block was already finalized, the block must be put in governance mode to be resolved be by hand
+(i.e. sudo or motion or other mechanics that are available ).
+
+As such the validator has to keep track of all votes irrespective if the disputed block is already known or not.
+All backing votes should be either kept in storage as well, or be queried on demand, since they are a kind of vote
+as well.
+
+## Late votes
+
+Late votes, after the dispute already reached a quorum + super majority, must be rewarded (albeit a smaller amount) as well.
+These ones must be attached to the votes after a defined period of time after the result has reached
+the required quorum + supermajority.
+
+## Chain Selection / Grandpa
+
+Chain selection should be influenced by the chance of picking a chain that does not even include the disputed block.
```

Comment:
I think we should avoid voting to finalize any chains that have an active dispute or have had a dispute resolve to the negative. This is combined with the voting rule of not finalizing anything which doesn't appear to have been approved by enough approval checkers.

Reply:
Not quite sure this section fits in here, or if that should be covered in a document itself and just be linked. It seems a bit out of scope for disputes and not affect the actual logic at all.